repositories/decisions_repository.go

```diff
@@ -489,31 +491,25 @@ func (repo *DecisionRepositoryImpl) channelOfDecisions(ctx context.Context, exec
 		defer close(decisionsChannel)
 		defer close(errChannel)
 
-		dbDecisionsChannel, dbErrChannel := SqlToChannelOfModels(ctx, exec, query, func(
-			row pgx.CollectableRow,
-		) (dbmodels.DbDecision, error) {
-			return pgx.RowToStructByName[dbmodels.DbDecision](row)
-		})
+		dbDecisionsChannel, dbErrChannel := SqlToChannelOfModels(
+			ctx,
+			exec,
+			query,
+			func(
+				row pgx.CollectableRow,
+			) (dbmodels.DbDecision, error) {
+				return pgx.RowToStructByName[dbmodels.DbDecision](row)
+			},
+		)
 
 		var allErrors []error
 
-		// Let's keep the non optimized version
-		// for dbDecision := range dbDecisionsChannel {
-		// 	rules, err := repo.rulesOfDecision(tx, dbDecision.Id)
-		// 	if err != nil {
-		// 		allErrors = append(allErrors, err)
-		// 		// do not send invalid decisions
-		// 		continue
-		// 	}
-		// 	decisionsChannel <- dbmodels.AdaptDecision(dbDecision, rules)
-		// }
-
 		for dbDecisions := range BatchChannel(dbDecisionsChannel, decisionRulesBatchSize) {
 
 			// fetch rules of all decisions
 			rules, err := repo.rulesOfDecisionsBatch(
 				ctx,
-				nil,
+				exec,
```

Comment:
missed that one last week - I tried a new research on "nil," and I think it's the last one